magento/magento2
# Refactor ConfigGenerator to replace/set crypt key instead of append

Fix the command "setup:config:set --key=..." to replace existing crypt key instead of append it.

### Description
Originally a call so "bin/magento setup:config:set --key='foobar'" did
not replace the crypt key, but append it with a newline.

This does not only "break" handling of values, encryptwed with the old
key, but also don't allow use of new key, because the new key is
composed of the old one, a newline and the new one.


### Fixed Issues (if relevant)
1. magento/magento2#11089: setup:config:set --key append instead of replace

### Manual testing scenarios
1. Install Magento & get sure, that there is a crypt key defined in env.php
2. Call "bin/magento setup:config:set --key=<new key>"
3. New crypt key in env.php should be your new key, instead of "<old>\n<new>"

### Contribution checklist
